src/apify/_memory_storage/resource_clients/dataset.py

```diff
@@ -188,47 +191,49 @@ async def list_items(
         )
 
         if existing_dataset_by_id is None:
-            _raise_on_non_existing_storage(_StorageTypes.DATASET, self._id)
+            raise_on_non_existing_storage(StorageTypes.DATASET, self._id)
 
-        async with existing_dataset_by_id._file_operation_lock:
-            start, end = existing_dataset_by_id._get_start_and_end_indexes(
-                max(existing_dataset_by_id._item_count - (offset or 0) - (limit or LIST_ITEMS_LIMIT), 0) if desc else offset or 0,
+        async with existing_dataset_by_id._file_operation_lock:  # type: ignore
+            start, end = existing_dataset_by_id._get_start_and_end_indexes(  # type: ignore
+                max(existing_dataset_by_id._item_count - (offset or 0) - (limit or LIST_ITEMS_LIMIT), 0) if desc else offset or 0,  # type: ignore
                 limit,
             )
 
             items = []
 
             for idx in range(start, end):
                 entry_number = self._generate_local_entry_name(idx)
-                items.append(existing_dataset_by_id._dataset_entries[entry_number])
+                items.append(existing_dataset_by_id._dataset_entries[entry_number])  # type: ignore
 
-            await existing_dataset_by_id._update_timestamps(False)
+            await existing_dataset_by_id._update_timestamps(has_been_modified=False)  # type: ignore
 
```

Comment:
so many type ignores here, why?

Reply:
Please continue with the discussion here https://github.com/apify/apify-sdk-python/pull/133#discussion_r1414245709.